### DISCUSSION TITLE
sixel doesn't work on windows

### DISCUSSION THREAD

**Discussion by VPraharsha03** [fastfetch-cli/fastfetch#1243] [Q&A]
Hi, recently sixel support landed to windows terminal, i tried to display logo with fastfetch but it doesn't work, i've used `--logo-type sixel` but it reverts to default logo and even tried using `chafa` to convert logo to sixels but it displays a very pixelated image and ruins the rendering

I've a couple questions:
- Does the release of fastfetch come with imagemagick pre-compiled ?
- Do i've to install a sixel backend on windows ? if so can you point me to a guide ?

**Comment by CarterLi** [comment]
> Does the release of fastfetch come with imagemagick pre-compiled ?

No.

I compiled fastfetch locally with imagemagick support. Sixel image does work for OpenConsole (A newer Conhost bundled in Windows Terminal). However it still doesn't work for Windows Terminal due to https://github.com/microsoft/terminal/issues/8581

![image](https://github.com/user-attachments/assets/09ef1623-a1c2-4a2e-9026-5ce38e3fb965)

**Reply by VPraharsha03** [reply]
Neofetch could display logo under WSL (tested both on OpenConsole & Windows Terminal) using sixel backend when I tested yesterday.

Sorry I've not gone through the above issue fully that you mentioned as it's too long, but it seems to prevent fastfetch from displaying logos, right ?

Would it be a good idea that releases come with imagemagick support ?

**Comment by CarterLi** [comment]
Okey. It turns out that https://github.com/microsoft/terminal/issues/8581 is supported, but has bugs. Did some trick and now sixel image protocol works for me in Windows Terminal

![image](https://github.com/user-attachments/assets/ad6c18a5-88a6-45b5-a1ec-bb562df16c05)

**Reply by VPraharsha03** [reply]
I tried compiling the latest as well as a fairly old version of fastfetch with `imagemagick` support but i'm unable to display logos on both WT & OpenConsole, here's my build log

[build_log.txt](https://github.com/user-attachments/files/16871369/build_log.txt)

I see a `??` or it reverts to the default logo

**Reply by kimusan** [reply]
What did you do to get it to work? I am not getting anything rendered in WSL2 in windows terminal 1.23.

**Comment by VPraharsha03** [comment]
@CarterLi would it be nice to include ootb support for imagemagick in every windows release ? what do you think ?

**Comment by Zeroes1** [comment]
@CarterLi 
Windows Terminal work fine with Sixel technology since latest Preview version : Microsoft.WindowsTerminalPreview_1.22.2362.0_x64.zip
or latest canary build.

![image](https://github.com/user-attachments/assets/6b23b0b6-51b7-44f1-9e92-1c0740076b83)

**Reply by CarterLi** [reply]
So what?

**Reply by wheatwhole** [reply]
hi I’m on 2025 terminal still doesn’t work please help

**Comment by CarterLi** [comment]
You can convert your image to sixel format by any online converter. Then use `fastfetch --raw /path/to/image.sixel --logo-width <width> --logo-height <height>` to display the image without needing imagemagick support of fastfetch

**Comment by Kalclavia** [comment]
@CarterLi I'm also looking for support for this. I am able to display native sixel images fine, but this is extremely annoying and finicky as I have to manually resize the image, text, and resize over any over.

I successfully compiled fastfetch locally to include imagemagick7 and chafa. I now get the error Logo: getCharacterPixelDimensions() failed when trying to display an .png as sixel, and this is due to the bug you linked above.

How were you able to patch this to get sixel working? If this is a relatively simple fix please let me know and I will do it myself locally.

**Reply by CarterLi** [reply]
> I now get the error Logo: getCharacterPixelDimensions() failed when trying to display an .png as sixel, and this is due to the bug you linked above.

It has been fixed in the dev branch

**Comment by Hi-ImKyle** [comment]
I checked the features on my fastfetch install and imagemagick is available, so I assumed `--sixel` would work. Does it need more than that? `fastfetch --sixel ~/V.png --logo-width 30` just sits there doing nothing. Should I create a Issue?

`fastfetch --raw ~/V.sixel --logo-width 30` works fine but it clears the terminal for some reason.. I assume since its raw?

**Reply by CarterLi** [reply]
> I checked the features on my fastfetch install and imagemagick is available

It was built with imagemagick features but imagemagick dlls are not bundled in the zip archives.

> but it clears the terminal for some reason..

You also need to set `--logo-height`

**Reply by Hi-ImKyle** [reply]
So what needs doing to get `--sixel` to work, is it just having the libs next to the fastfetch binary?

And the height ok, my bad.

**Reply by VPraharsha03** [reply]
> So what needs doing to get `--sixel` to work, is it just having the libs next to the fastfetch binary?

@CarterLi i've added image magick dir (having libs) to PATH and set MAGICK_HOME but still i get image magick library not found

**Reply by CarterLi** [reply]
You may install fastfetch in MSYS2 if you want sixel support. For example: `pacman -S mingw-w64-clang-x86_64-fastfetch mingw-w64-clang-x86_64-imagemagick`

**Reply by VPraharsha03** [reply]
i'm giving up on sixel on windows

i'm unable to find the right logo height & width with `--raw`, the image either just cuts through the text or text gets pushed far away from image; it's surprising to see that changing height & width doesn't impact logo but rather changes the alignment of text (bug?) 

tried clear before fastfetch, tried padding but that didn't work as well

**Reply by 0mn1core** [reply]
I've gotten sixel to work and gotten the height/width set so that the text is aligned with the image. The only problem I can't figure out how to solve now is that there's a bunch of whitespace after the fastfetch command that I have no idea how to remove. I'm assuming it's making space for the text being underneath the image, and just keeps that space even if the text gets correctly aligned with the image?
<img width="1502" height="1010" alt="image" src="https://github.com/user-attachments/assets/d58e5c0a-e2a5-4c57-a565-ffd8e926aa25" />
Messing around with a modification of theme found here: https://github.com/LierB/fastfetch

Update: Messing around with it, having both height and width to logo seems to result in all this extra space, no matter what value for dimensions, what type I use, and whether it loads the image or falls back to the default ascii. Is this a known bug?

**Reply by VPraharsha03** [reply]
@0mn1core I've had a similar issue, will try updating fastfetch and Windows Terminal to see. I see this happening only on Windows, on linux it works perfectly. I might open an issue.

**Reply by Kalclavia** [reply]
@0mn1core If possible, try only specifying the height or playing with the dimensions parameters. Haven't looked at this in a while, but I remember seeing this and that specifying the width is what caused the extra space. Either way, this probably warrants a fix.

**Reply by VPraharsha03** [reply]
I tried on latest versions of both stable and canary, in my case only `--logo-width` works and gives output well, but ironically `--logo-height` or `--logo-width` doesn't resize the actual sixel image and instead shifts the position of text

<img width="1920" height="942" alt="fastfetch" src="https://github.com/user-attachments/assets/9fb17cfa-95c9-4644-bbbd-2c3b5c505f07" />

specifying width lesser than the image's size results in text cutting through the image:

<img width="1397" height="802" alt="ff2" src="https://github.com/user-attachments/assets/868ad924-45b8-4288-9224-05bd759c990a" />

**Comment by Kalclavia** [comment]
I got this working briefly. I think the canary version of windows terminal
supports sixel, or one of the other formats used by fast fetch. I also had
to explicitly tell fast fetch to use that format. It wasn’t perfect but it
did work, I think the width and height constraints didn’t work right.

-------
Very Respectfully,

Abhi Bhattacharyya
